platform/centec-arm64/platform.conf

```diff
@@ -0,0 +1,135 @@
+	#cd $demo_mount/boot/
+	#mv vmlinuz* Image
+	#gzip -c Image > Image.gz
+	#mkimage -f centec-e530.its centec-e530.itb
+	#cd -
+        cp $demo_mount/boot/sonic_arm64.fit /mnt/flash/boot/centec-e530.itb
+        cd /mnt/flash/boot
+        rm onie_uimage -rf
+        ln -s centec-e530.itb onie_uimage
+        cd -
+    else
+        rm /mnt/flash/boot/centec-e550.itb -rf
+        cp $demo_mount/boot/centec-e550.itb /mnt/flash/boot
+        cd /mnt/flash/boot
+        rm onie_uimage -rf
+        ln -s centec-e550.itb onie_uimage
+        cd -
+    fi
+
+    # Mount debian rootfs
+    losetup /dev/loop0 $demo_mount/fs.squashfs
+    mount /dev/loop0 $demo_mount/.overlay_ro
+    mount -t overlay overlay -o lowerdir=/mnt/flash/debian/.overlay_ro,upperdir=/mnt/flash/debian/.overlay_rw,workdir=/mnt/flash/debian/.overlay_work /mnt/flash/debian/.rootfs_dir
+
+    # unzip dockerfs to sonic debian roootfs
+    if [ ! -d $demo_mount/.rootfs_dir/var/lib/$DOCKERFS_DIR ]; then
+        mkdir $demo_mount/.rootfs_dir/var/lib/$DOCKERFS_DIR
+    fi
+    mount --bind $demo_mount/.docker /mnt/flash/debian/.rootfs_dir/var/lib/$DOCKERFS_DIR
+    unzip -op $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" | tar -xpz -f - -C $demo_mount/.docker
+
+    # copy config files to rootfs
+    echo 'machine=centec-arm64' > /mnt/flash/debian/.rootfs_dir/host/machine.conf
+    fw_printenv | grep '^onie_platform=' >> /mnt/flash/debian/.rootfs_dir/host/machine.conf
+    SONIC_VERSION=`chroot /mnt/flash/debian/.rootfs_dir /usr/local/bin/sonic-cfggen -y /etc/sonic/sonic_version.yml -v build_version`
+    chroot /mnt/flash/debian/.rootfs_dir mkdir -p /host/image-$SONIC_VERSION/
+    cp $demo_mount/platform /mnt/flash/debian/.rootfs_dir/host/image-$SONIC_VERSION/ -rf
+
+    # install centec platform modules
+    #PKG_NAME="platform-modules-"`fw_printenv | grep ^onie_machine= | awk -F _ '{print $3 "-" $4}'`
+    #PKG_DIR="/platform/arm64-"`fw_printenv | grep ^onie_machine= | awk -F = '{print $2}'`"-r0"
+    #DEB_NAME=$PKG_NAME"_1.1_arm64.deb"
+    #chroot /mnt/flash/debian/.rootfs_dir dpkg -i $PKG_DIR/$DEB_NAME
+
+    # sync filesystem
+    sync
+    umount -l /mnt/flash/boot
+    umount -l /mnt/flash
+    sync
+    sync
+
+    echo "Reboot board to boot from installed OS"
+}
+
+hw_load() {
+    echo "ext4load mmc 0:2 \$loadaddr onie_uimage && run onie_args && bootm"
+}
+
+
+install_uimage
+
+hw_load_str="$(hw_load)"
+
+echo "Updating U-Boot environment variables"
+(cat <<EOF
+hw_load $hw_load_str
+copy_img echo "Loading Demo $platform image..." && run hw_load
+nos_bootcmd run copy_img && setenv bootargs quiet console=\$consoledev,\$baudrate && bootm \$loadaddr
+EOF
+) > /tmp/env.txt
+
+fw_setenv -f -s /tmp/env.txt
+
+cd /
+
+# Set NOS mode if available.  For manufacturing diag installers, you
+# probably want to skip this step so that the system remains in ONIE
+# "installer" mode for installing a true NOS later.
+if [ -x /bin/onie-nos-mode ] ; then
+    /bin/onie-nos-mode -s
+fi
```

Comment:
i can see duplicated code available in installer/arm64/installer.sh

can you reuse their code by not duplicating those code here?

Reply:
OK, code will be optimized, thank you.